### ISSUE TITLE
[DOC] documentation is not clear and implied that PyT Trainer class  cannot support BC and Regression

### ISSUE THREAD

**Issue by rnyak** [NVIDIA-Merlin/Transformers4Rec#414]
We need to clarify in the documentation that PyT Trainer class  cannot support BC and Regression. We should explain what users can use  for these tasks instead of Trainer Class : they can use custom developed `model.fit()` and `model.evaluate()`.

**Comment by rnyak** [comment]
@sararb with the new PRs do we add any explanation in the docs?

**Comment by sararb** [comment]
Yes, It is planned to add the documentation about how to train BC/regression with the HF Trainer class after the task https://github.com/NVIDIA-Merlin/Transformers4Rec/issues/554 is finalized.
